node/network/statement-distribution/src/lib.rs

```diff
@@ -831,94 +847,132 @@ async fn handle_network_update(
+							);
+
+							ctx.send_messages(
+								std::iter::once(val_message).chain(std::iter::once(session_message))
+							).await?;
+
+							match (val_rx.await?, session_rx.await?) {
+								(Ok(v), Ok(s)) => (v, s),
+								(Err(e), _) | (_, Err(e)) => {
+									log::warn!(
+										target: LOG_TARGET,
+										"Failed to fetch runtime API data for active leaf: {:?}",
+										e,
+									);
+
+									// Lacking this bookkeeping might make us behave funny, although
+									// not in any slashable way. But we shouldn't take down the node
+									// on what are likely spurious runtime API errors.
+									continue;
+								}
 							}
-						}
-					};
+						};
 
-					active_heads.entry(relay_parent)
-						.or_insert(ActiveHeadData::new(validators, session_index));
+						active_heads.entry(relay_parent)
+							.or_insert(ActiveHeadData::new(validators, session_index));
+					}
 				}
-			}
-			FromOverseer::Signal(OverseerSignal::BlockFinalized(_block_hash)) => {
-				// do nothing
-			}
-			FromOverseer::Signal(OverseerSignal::Conclude) => break,
-			FromOverseer::Communication { msg } => match msg {
-				StatementDistributionMessage::Share(relay_parent, statement) => {
-					inform_statement_listeners(
-						&statement,
-						&mut statement_listeners,
-					).await;
-					circulate_statement_and_dependents(
-						&mut peers,
-						&mut active_heads,
-						&mut ctx,
-						relay_parent,
-						statement,
-					).await?;
+				FromOverseer::Signal(OverseerSignal::BlockFinalized(_block_hash)) => {
+					// do nothing
 				}
-				StatementDistributionMessage::NetworkBridgeUpdateV1(event) =>
-					handle_network_update(
-						&mut peers,
-						&mut active_heads,
-						&mut ctx,
-						&mut our_view,
-						event,
-					).await?,
-				StatementDistributionMessage::RegisterStatementListener(tx) => {
-					statement_listeners.push(tx);
+				FromOverseer::Signal(OverseerSignal::Conclude) => break,
+				FromOverseer::Communication { msg } => match msg {
+					StatementDistributionMessage::Share(relay_parent, statement) => {
+						inform_statement_listeners(
+							&statement,
+							&mut statement_listeners,
+						).await;
+						circulate_statement_and_dependents(
+							&mut peers,
+							&mut active_heads,
+							&mut ctx,
+							relay_parent,
+							statement,
+							&metrics,
+						).await?;
+					}
+					StatementDistributionMessage::NetworkBridgeUpdateV1(event) =>
+						handle_network_update(
+							&mut peers,
+							&mut active_heads,
+							&mut ctx,
+							&mut our_view,
+							event,
+							&metrics,
+						).await?,
+					StatementDistributionMessage::RegisterStatementListener(tx) => {
+						statement_listeners.push(tx);
+					}
 				}
 			}
 		}
+		Ok(())
+	}
+}
+
+#[derive(Clone)]
+struct MetricsInner {
+	statements_distributed: prometheus::Counter<prometheus::U64>,
+}
+
+/// Statement Distribution metrics.
+#[derive(Default, Clone)]
+pub struct Metrics(Option<MetricsInner>);
```

Comment:
Earlier it was made named member, instead of a newtype. A bit inconsistent.

Reply:
Hmm, I think it's a newtype everywhere now, could you point me to a named member?

Reply:
Hmm, I think what I saw was not a `Metrics` definition but a unit `struct` that was extended with a `Metrics` member named `metrics`.
Discard this comment.